src/audio/src/src.c

```diff
@@ -576,6 +576,7 @@ static void src_set_sink_params(struct comp_dev *dev, struct sof_sink __sparse_c
 	sink_set_valid_fmt(sink, valid_fmt);
 	sink_set_channels(sink, cd->ipc_config.base.audio_fmt.channels_count);
 	sink_set_buffer_fmt(sink, cd->ipc_config.base.audio_fmt.interleaving_style);
+	sink_set_rate(sink, cd->ipc_config.sink_rate);
```

Comment:
possible to have a common function like: ipc4_update_buffer_format to cover all set in one function then can be used for all modules?

Reply:
that would need to be done at the module layer. 1st step needs to be aliging all module to use the module API, 2nd step would be to then inpect each module to see where we can move repeating logic from modules to core.